packages/data/src/world/mm/great_bay_temple.yml

```diff
@@ -180,14 +180,14 @@
   locations:
     "Great Bay Temple Green Pipe 2 Lower Chest": "can_hookshot || (can_use_ice_arrows && can_hookshot_short)" #can just jump to the alcove after riding the platform, new trick?
     "Great Bay Temple Green Pipe 2 Upper Chest": "(can_hookshot && can_use_ice_arrows && (can_use_fire_arrows || trick(MM_GBT_FIRELESS))) || short_hook_anywhere"
-    "Great Bay Temple Pot Green Pipe 2 1": "underwater_walking || hookshot_anywhere || (can_use_ice_arrows && short_hook_anywhere)"
-    "Great Bay Temple Pot Green Pipe 2 2": "underwater_walking || hookshot_anywhere || (can_use_ice_arrows && short_hook_anywhere)"
-    "Great Bay Temple Pot Green Pipe 2 3": "underwater_walking || hookshot_anywhere || (can_use_ice_arrows && short_hook_anywhere)"
-    "Great Bay Temple Pot Green Pipe 2 4": "underwater_walking || hookshot_anywhere || (can_use_ice_arrows && short_hook_anywhere)"
-    "Great Bay Temple Pot Green Pipe 2 5": "underwater_walking || hookshot_anywhere || (can_use_ice_arrows && short_hook_anywhere)"
-    "Great Bay Temple Pot Green Pipe 2 6": "underwater_walking || hookshot_anywhere || (can_use_ice_arrows && short_hook_anywhere)"
-    "Great Bay Temple Pot Green Pipe 2 7": "underwater_walking || (can_use_ice_arrows && short_hook_anywhere)"
-    "Great Bay Temple Pot Green Pipe 2 8": "underwater_walking || (can_use_ice_arrows && short_hook_anywhere)"
+    "Great Bay Temple Pot Green Pipe 2 1": "(can_dive_small && (has_weapon_range || has_bombchu)) || underwater_walking || hookshot_anywhere || (can_use_ice_arrows && short_hook_anywhere)"
+    "Great Bay Temple Pot Green Pipe 2 2": "(can_dive_small && (has_weapon_range || has_bombchu)) || underwater_walking || hookshot_anywhere || (can_use_ice_arrows && short_hook_anywhere)"
+    "Great Bay Temple Pot Green Pipe 2 3": "(can_dive_small && (has_weapon_range || has_bombchu)) || underwater_walking || hookshot_anywhere || (can_use_ice_arrows && short_hook_anywhere)"
+    "Great Bay Temple Pot Green Pipe 2 4": "(can_dive_small && (has_weapon_range || has_bombchu)) || underwater_walking || hookshot_anywhere || (can_use_ice_arrows && short_hook_anywhere)"
+    "Great Bay Temple Pot Green Pipe 2 5": "(can_dive_small && (has_weapon_range || has_bombchu)) || underwater_walking || hookshot_anywhere || (can_use_ice_arrows && short_hook_anywhere)"
+    "Great Bay Temple Pot Green Pipe 2 6": "(can_dive_small && (has_weapon_range || has_bombchu)) || underwater_walking || hookshot_anywhere || (can_use_ice_arrows && short_hook_anywhere)"
+    "Great Bay Temple Pot Green Pipe 2 7": "(can_dive_small && (has_weapon_range || has_bombchu)) || underwater_walking || (can_use_ice_arrows && short_hook_anywhere)"
+    "Great Bay Temple Pot Green Pipe 2 8": "(can_dive_small && (has_weapon_range || has_bombchu)) || underwater_walking || (can_use_ice_arrows && short_hook_anywhere)"
```

Comment:
All of these will need to check specifically for arrows or hook, the last two only arrows since hook cannot reach them.

Reply:
Ah yeah, or ice arrows.